File: src/csharp/Intel/Iced/Intel/CodeWriter.cs
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

#if ENCODER
namespace Iced.Intel {
	/// <summary>
	/// Used by an <see cref="Encoder"/> to write encoded instructions
	/// </summary>
	public abstract class CodeWriter {
		/// <summary>
		/// Writes the next byte
		/// </summary>
		/// <param name="value">Value</param>
		public abstract void WriteByte(byte value);
	}
}
#endif

File: src/java/iced-x86/src/main/java/com/github/icedland/iced/x86/CodeWriter.java
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

package com.github.icedland.iced.x86;

/**
 * Used by an {@link com.github.icedland.iced.x86.enc.Encoder} to write encoded instructions
 */
public interface CodeWriter {
	/**
	 * Writes the next byte
	 *
	 * @param value Value
	 */
	public abstract void writeByte(byte value);
}
